### PR TITLE
feat(widget-chat): add the chat widget feature to packages

### PR DESCRIPTION
**Pull request by adamweeks**
This will be the first "official" merge of the chat widget into the master branch.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,4 @@ _site
 string
 .sauce.tid
 .npm
+local/
```
Comment (ianwremmel) [inline .gitignore]: what goes in `local`?

File: .stylelintrc [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    "selector-no-qualifying-type": true,
+    "selector-no-universal": true,
+    "font-weight-notation": "named-where-possible",
+    "no-browser-hacks": true,
+    "property-no-unknown": [ true, {
+      "ignoreProperties": ["composes"]
+    }],
+    "declaration-block-properties-order": [
+      "composes"
+    , "position"
```
Comment (ianwremmel) [inline .stylelintrc]: leading commas? Really @bzang? :)

Seriously though, does stylelint support yml files?

File: .gitignore (hunk 1 above)
Comment (bzang) [inline .gitignore]: We used it as a reference for a background file for the demo. I don't think we need it at this point.

File: .stylelintrc (hunk 2 above)
Comment (bzang) [inline .stylelintrc]: Ahhh seems like it might. We'll have to test this out. Mah bad on the formatting there.

**Pull request closed without merging**